src/components/app.jsx

```diff
@@ -1,9 +1,163 @@
 import React from "react";
-import { ReactComponent as Logo } from "../images/sun.svg";
 import "../App.css";
+import { getWeather, getForecast } from '../utils.js';
+import Header from './Header';
+import WeatherCurrent from './WeatherCurrent';
+import WeatherForecast from './WeatherForecast';
 
-function App() {
-  return <h1>Weather</h1>;
+class App extends React.Component {
+  state = {
```

Comment:
Holding all of the state and handling mutations at the App root level is workable for a simple app like this. If we had to scale out the feature-set here, what strategies might you consider for refactoring the data flow?

Reply:
Typically, on larger apps, if a child component needs to be stateful, I consider whether the data is necessary for just for that component. If it is, then I store that data in the child component's state instead of in the root component's state. For example, I usually make `<Login />` components stateful, as the data and methods needed there aren't really necessary in the rest of the app. To my understanding, this process could be even further refactored with React Redux or Context API, as they should be able to separate the state management from the presentation logic; however, I don't currently have experience with either of those (would love to learn though!).